pkg/queue/queue.go

```diff
@@ -9,21 +9,23 @@ import (
 	"time"
 )
 
-func New(initialConcurrency int) *Queue {
+func New(initialConcurrency int, concurrencyPerReplica uint32) *Queue {
 	return &Queue{
-		concurrency: initialConcurrency,
-		list:        list.New(),
-		enqueued:    make(chan struct{}),
-		completed:   make(chan struct{}),
+		concurrency:           initialConcurrency,
+		concurrencyPerReplica: concurrencyPerReplica,
+		list:                  list.New(),
+		enqueued:              make(chan struct{}),
+		completed:             make(chan struct{}),
 	}
 }
 
 // Queue is a thread-safe FIFO queue that will limit the number of concurrent
 // requests that can be in progress at any given time.
 type Queue struct {
-	// concurrency is the max number of in progress items.
-	concurrency    int
 	concurrencyMtx sync.RWMutex
+	// concurrency is the max number of in progress items.
+	concurrency           int
+	concurrencyPerReplica uint32
```

Comment:
please add a comment for `concurrencyPerReplica` so people reading directly understand what it does.

e.g. add 
```
// concurrencyPerReplica is the max number of in progress items per replica.
```